google/resource_logging_project_sink_test.go

```diff
@@ -0,0 +1,203 @@
+
+func TestAccLoggingProjectSink_uniqueWriter(t *testing.T) {
+	sinkName := "tf-test-sink-" + acctest.RandString(10)
+	bucketName := "tf-test-sink-bucket-" + acctest.RandString(10)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLoggingProjectSinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLoggingProjectSink_uniqueWriter(sinkName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectSink(
+						"google_logging_project_sink.unique_writer"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccLoggingProjectSink_updateUniqueWriter(t *testing.T) {
+	sinkName := "tf-test-sink-" + acctest.RandString(10)
+	bucketName := "tf-test-sink-bucket-" + acctest.RandString(10)
+	updatedBucketName := "tf-test-sink-bucket-" + acctest.RandString(10)
+
+	var sinkBefore, sinkAfter logging.LogSink
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLoggingProjectSinkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLoggingProjectSink_uniqueWriter(sinkName, bucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectSinkExists("google_logging_project_sink.unique_writer", &sinkBefore),
+					testAccCheckLoggingProjectSink(
+						"google_logging_project_sink.unique_writer"),
+				),
+			}, {
+				Config: testAccLoggingProjectSink_uniqueWriterUpdated(sinkName, updatedBucketName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLoggingProjectSinkExists("google_logging_project_sink.unique_writer", &sinkAfter),
+					testAccCheckLoggingProjectSink(
+						"google_logging_project_sink.unique_writer"),
+				),
+			},
+		},
+	})
+
+	// Destination and Filter should have changed, but WriterIdentity should be the same
+	if sinkBefore.Destination == sinkAfter.Destination {
+		t.Errorf("Expected Destination to change, but it didn't: Destination = %#v", sinkBefore.Destination)
+	}
+	if sinkBefore.Filter == sinkAfter.Filter {
+		t.Errorf("Expected Filter to change, but it didn't: Filter = %#v", sinkBefore.Filter)
+	}
+	if sinkBefore.WriterIdentity != sinkAfter.WriterIdentity {
+		t.Errorf("Expected WriterIdentity to be the same, but it differs: before = %#v, after = %#v",
+			sinkBefore.WriterIdentity, sinkAfter.WriterIdentity)
+	}
+}
+func testAccCheckLoggingProjectSinkDestroy(s *terraform.State) error {
+	config := testAccProvider.Meta().(*Config)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "google_logging_project_sink" {
+			continue
+		}
+
+		attributes := rs.Primary.Attributes
+
+		_, err := config.clientLogging.Projects.Sinks.Get(attributes["id"]).Do()
+		if err == nil {
+			return fmt.Errorf("project sink still exists")
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckLoggingProjectSinkExists(n string, sink *logging.LogSink) resource.TestCheckFunc {
```

Comment:
Since everything this function does is also in the next one, is it worth it to keep them separate?

Reply:
I kept them separate for now since `testAccCheckLoggingProjectSinkExists` assigns the found api object to a provided `*logging.LogSink` parameter and `testAccCheckLoggingProjectSink` checks the remote object against the locally stored attributes

However, since they both look up the api object I changed `testAccCheckLoggingProjectSink` to take an api object as the first parameter; now it does less duplicate work